ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh

```diff
@@ -47,15 +47,28 @@ function mirror_release_image_for_disconnected_upgrade() {
       ssh "${SSHOPTS[@]}" "root@${IP}" bash - << EOF
 MIRRORED_RELEASE_IMAGE=${DS_REGISTRY}/localimages/local-release-image
 DIGEST=\$(oc adm release info --registry-config ${DS_WORKING_DIR}/pull_secret.json ${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE} --output=jsonpath="{.digest}")
+RELEASE_TAG=\$(sed -e "s/^sha256://" <<< \${DIGEST})
+MIRROR_RESULT_LOG=/tmp/image_mirror-\${RELEASE_TAG}.log
+
 echo "Mirroring release images for disconnected environment"
-oc adm release mirror --registry-config ${DS_WORKING_DIR}/pull_secret.json --from=${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE} --to=\${MIRRORED_RELEASE_IMAGE} --apply-release-image-signature
-echo "OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE=\${MIRRORED_RELEASE_IMAGE}@\${DIGEST}" >> /tmp/disconnected_mirror.conf
+oc adm release mirror --registry-config ${DS_WORKING_DIR}/pull_secret.json \
+  --from=${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE} \
+  --to=\${MIRRORED_RELEASE_IMAGE} \
+  --to-release-image=\${MIRRORED_RELEASE_IMAGE}:\${RELEASE_TAG}  2>&1 | tee \${MIRROR_RESULT_LOG}
+
+echo "Create ImageContentSourcePolicy to use mirrored registry in upgrade"
+UPGRADE_ICS=\$(cat \${MIRROR_RESULT_LOG} | sed -n '/repositoryDigestMirrors/,//p')
+
+cat <<EOF1 | oc apply -f -
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: disconnected-upgrade-ics
+spec:
+\${UPGRADE_ICS}
+EOF1
 EOF
 
-      # shellcheck source=/dev/null
-      source <(ssh "${SSHOPTS[@]}" "root@${IP}" "cat /tmp/disconnected_mirror.conf")
-      echo "OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE is overridden to ${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"
```

Comment:
Could we keep this echo? I guess it would be useful when troubleshooting

Reply:
We are not overriding image anymore

Reply:
We are using actual image as upgrade and it is automatically mirrored to our registry via `ImageContentSourcePolicy`

Reply:
That's the canonical env var name used for injecting the [release:latest pullspec](https://github.com/openshift/release/blob/ecc2740af58bad9e0e96f70fdb6a97b270f52c94/ci-operator/step-registry/baremetalds/e2e/upgrade/baremetalds-e2e-upgrade-workflow.yaml#L12) and still used: 
 ```
oc adm release mirror --registry-config ${DS_WORKING_DIR}/pull_secret.json \
  --from=${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE} \
```
The echo will be just a tip in the log to specify to which version we're currently upgrading to

Reply:
But since we are not overriding, it is in;

```
openshift-tests run-upgrade all --to-image registry.build03.ci.openshift.org/ci-op-3g123y5y/release@sha256:5d19aec75540e8056cca0ab3ab1ee9c4098c0a703b1264ded57a477cae0c496e --provider '{"type":"baremetal","disconnected":true}' --from-repository virthost.ostest.test.metalkube.org:5000/localimages/local-test-image -o /logs/artifacts/e2e.log --junit-dir /logs/artifacts/junit
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/25130/rehearse-25130-pull-ci-openshift-cluster-baremetal-operator-master-e2e-metal-ipi-upgrade-ovn-ipv6-trigger/1481247658726133760/artifacts/e2e-metal-ipi-upgrade-ovn-ipv6-trigger/baremetalds-e2e-test/build-log.txt

Reply:
I think the var name it's a bit unfortunate, anyhow I noticed that the info I was really looking for is also reported in a better way later, ie: 

```
"desired": {
      "version": "4.10.0-0.ci.test-2022-01-12-125752-ci-op-3g123y5y-latest",
      "image": "registry.build03.ci.openshift.org/ci-op-3g123y5y/release@sha256:5d19aec75540e8056cca0ab3ab1ee9c4098c0a703b1264ded57a477cae0c496e"
    }
```